File: Algorithms/Sorting/Bubble_Sort/Bubble_Sort.js
function swap(arr, first_Index, second_Index){
    var temp = arr[first_Index];
    arr[first_Index] = arr[second_Index];
    arr[second_Index] = temp;
}

function bubble_Sort(arr){
  var len = arr.length,
        i, j, stop;
    for (i=0; i < len; i++){
        for (j=0, stop=len-i; j < stop; j++){
            if (arr[j] > arr[j+1]){
                swap(arr, j, j+1);
            }
        }
    }
    return arr;
}
console.log(bubble_Sort([10, 5, 2, 5, -6, 4, -1])); //array that is to be sorted

/*
algorithm
function swap( array, first_Element_index, second_element_index){
   swap the variables using temporary variable
}
function bubble_Sort(array)
{
len=array.length
 for i = 0 to len :
      swapped = false
		
      for j = 0 to len-i:
         (compare the adjacent elements)  
         if array[j] > array[j+1] then
            /* swap them */
            swap( array,j,j+1 )		 
            swapped = true
         end if
         
      end for
return arr
}    
pass the argument array
*/  

File: Algorithms/Sorting/Bubble_Sort/Bubble_Sort.java
import java.util.*;
import java.io.*;

class Bubble_Sort 
{ 
    void bubbleSort(int arr[]) 
    { 
        int n = arr.length; 
        for (int i = 0; i < n-1; i++) 
            for (int j = 0; j < n-i-1; j++) 
                if (arr[j] > arr[j+1]) 
                {
                    int temp = arr[j]; 
                    arr[j] = arr[j+1]; 
                    arr[j+1] = temp; 
                } 
    } 

    void printArray(int arr[]) 
    { 
        int n = arr.length; 
        for (int i=0; i<n; ++i) 
            System.out.print(arr[i] + " "); 
        System.out.println(); 
    } 

    public static void main(String args[]) 
    { 
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter array length : ");
        int n = sc.nextInt();
        int arr[] = new int[n];
        System.out.println("Enter array elements : ");
        
        for (int i=0;i<n ;i++ ) {
            arr[i] = sc.nextInt();
        }

        Bubble_Sort bs = new Bubble_Sort(); 
        bs.bubbleSort(arr); 
        System.out.println("Sorted array"); 
        bs.printArray(arr); 
    } 
} 